Source/stores.cpp

```diff
@@ -1580,18 +1320,19 @@ void StoreSellItem()
 	}
 }
 
-void SmithSellEnter()
+void SellEnter()
 {
 	if (stextsel == BackButtonLine()) {
-		StartStore(STORE_SMITH);
+		StartStore(stextshold);
 		stextsel = 16;

```

Comment:
Going back with the `Back` button didn't work correctly.
Example:
I tried to sell a item, but didn't have anything for the current store.
When I pressed enter, i expected tor return to the previous menu.
But instead I was asked if I want to sell a (not existing) item.

```suggestion
		StoreESC();
```

![grafik](https://user-images.githubusercontent.com/25415264/145714588-0b1cfb5b-fde3-43ed-9cd8-95b901ffe904.png)